pkg/controller/cassandracluster/reconcile.go

```diff
@@ -625,6 +670,9 @@ func FlipCassandraClusterUpdateSeedListStatus(cc *api.CassandraCluster, status *
 					logrus.WithFields(logrus.Fields{"cluster": cc.Name,
 						"dc-rack": dcRackName}).Infof("Update Rack Status UpdateSeedList=ToDo")
 					dcRackStatus.CassandraLastAction.Name = api.ActionUpdateSeedList
+					ClusterAction.With(
+						prometheus.Labels{"cluster": cc.Name},
+					).Set(api.ClusterActionVal(api.ActionUpdateSeedList))
```

Comment:
I wonder if this call could not be a more simple function such as : 

```go
func (clusterAction *GaugeVec) updateClusterAction(clusterName string, action Action) {
  clusterAction.With(prometheus.Labels{"cluster": clusterName},)
    .Set(api.ClusterActionVal(string(action)))
}
```

To make the code more readable.

In the same way, I think we need to change the Action, ClusterPhase, Status and Operation type, to something more restrictive than a String. Just as they did it in this operator : https://github.com/banzaicloud/kafka-operator/blob/master/api/v1beta1/common_types.go#L76

It makes the code more readable, and usage simplier.

Reply:
Okay for the first part. For the second part I don't know how difficult it would be as it ends in the status of the object. I need to take a closer look to the kafka operator I guess. Let's chat about it tomorrow